farm/modeling/biadaptive_model.py

```diff
@@ -0,0 +1,537 @@
+                else:
+                    raise ValueError(
+                        "Unknown extraction strategy from DPR model: {}".format(lm2_out)
+                    )
+
+                # Do the actual forward pass of a single head
+                all_logits.append(head(output1, output2))
+        else:
+            # just return LM output (e.g. useful for extracting embeddings at inference time)
+            all_logits.append((pooled_output1, pooled_output2))
+
+        return all_logits
+
+    def forward_lm(self, **kwargs):
+        """
+        Forward pass for the BiAdaptive model.
+
+        :param kwargs:
+        :return: 2 tensors of pooled_output from the 2 language models
+        """
+        pooled_output1, hidden_states1 = self.language_model1(**kwargs)
+        pooled_output2, hidden_states2 = self.language_model2(**kwargs)
+
+        return pooled_output1, pooled_output2
+
+    def log_params(self):
+        """
+        Logs paramteres to generic logger MlLogger
+        """
+        params = {
+            "lm1_type": self.language_model1.__class__.__name__,
+            "lm1_name": self.language_model1.name,
+            "lm1_output_types": ",".join(self.lm1_output_types),
+            "lm2_type": self.language_model2.__class__.__name__,
+            "lm2_name": self.language_model2.name,
+            "lm2_output_types": ",".join(self.lm2_output_types),
+            "prediction_heads": ",".join(
+                [head.__class__.__name__ for head in self.prediction_heads]
+            ),
+        }
+        try:
+            MlLogger.log_params(params)
+        except Exception as e:
+            logger.warning(f"ML logging didn't work: {e}")
+
+    def verify_vocab_size(self, vocab_size1, vocab_size2):
+        """ Verifies that the model fits to the tokenizer vocabulary.
+        They could diverge in case of custom vocabulary added via tokenizer.add_tokens()"""
+
+        model1_vocab_len = self.language_model1.model.resize_token_embeddings(new_num_tokens=None).num_embeddings
+
+        msg = f"Vocab size of tokenizer {vocab_size1} doesn't match with model {model1_vocab_len}. " \
+              "If you added a custom vocabulary to the tokenizer, " \
+              "make sure to supply 'n_added_tokens' to LanguageModel.load() and BertStyleLM.load()"
+        assert vocab_size1 == model1_vocab_len, msg
+
+        model2_vocab_len = self.language_model2.model.resize_token_embeddings(new_num_tokens=None).num_embeddings
+
+        msg = f"Vocab size of tokenizer {vocab_size1} doesn't match with model {model2_vocab_len}. " \
+              "If you added a custom vocabulary to the tokenizer, " \
+              "make sure to supply 'n_added_tokens' to LanguageModel.load() and BertStyleLM.load()"
+        assert vocab_size2 == model2_vocab_len, msg
+
+    def get_language(self):
+        return self.language_model1.language, self.language_model2.language
+
+    def convert_to_transformers(self):
```

Comment:
The conversion changed in a recent PR (https://github.com/deepset-ai/FARM/pull/576). We moved most of the logic to a separate Converter class. I think we should refactor this for DPR in a similar way, but we can do this in a separate PR.